src/main/java/com/volasoftware/tinder/services/implementations/AuthenticationServiceImpl.java

```diff
@@ -26,111 +27,130 @@
-        authenticationManager.authenticate(
-                new UsernamePasswordAuthenticationToken(
-                        loginRequest.getEmail(),
-                        loginRequest.getPassword()
-                )
-        );
+    return accountMapper.accountToAccountDto(updatedAccount);
+  }
 
-        Account account = accountRepository.findOneByEmail(loginRequest.getEmail())
-                .orElseThrow(() -> new AccountNotFoundException(AccountConstant.NOT_FOUND));
+  @Override
+  public LoginResponse login(LoginRequest loginRequest) {
+    authenticationManager.authenticate(
+        new UsernamePasswordAuthenticationToken(
+            loginRequest.getEmail(),
+            loginRequest.getPassword()
+        )
+    );
 
-        if (!account.isVerified()) {
-            throw new AccountNotVerifiedException(AccountConstant.NOT_VERIFIED);
-        }
+    Account account = findAccountByEmail(loginRequest.getEmail());
 
-        return new LoginResponse(jwtService.generateToken(account));
+    if (!account.isVerified()) {
+      throw new AccountNotVerifiedException(AccountConstant.NOT_VERIFIED);
     }
 
-    private void sendVerificationMail(String receiver, String token) {
-        String link = baseUrl + verifyUrl + token;
-        byte[] contentBytes = fileService.readHtml(FilePathConstant.VERIFICATION_EMAIL_HTML);
-        String content = new String(contentBytes).replace(HtmlConstant.HREF, link);
+    return new LoginResponse(jwtService.generateToken(account));
+  }
 
-        emailService.send(receiver, content);
+  @Override
+  public AccountDto updateAccount(AccountDto accountDto, Principal principal) {
+    Account account = findAccountByEmail(accountDto.getEmail());
+    if (!account.getEmail().equals(principal.getName())) {
+      throw new AccountNotOwnerException(AccountConstant.NOT_OWNER);
     }
 
-    private void checkIfEmailIsTaken(String email) {
-        Optional<Account> optionalAccount = accountRepository.findOneByEmail(email);
-        if (optionalAccount.isPresent()) {
-            throw new EmailIsTakenException(MailConstant.ALREADY_TAKEN);
-        }
+    account.setFirstName(accountDto.getFirstName());
+    account.setLastName(accountDto.getLastName());
+    account.setEmail(accountDto.getEmail());
+    account.setGender(accountDto.getGender());
+
+    return accountMapper.accountToAccountDto(accountRepository.save(account));
+  }
+
+  private Account findAccountByEmail(String email) {
+    return accountRepository.findOneByEmail(email)
+        .orElseThrow(() -> new AccountNotFoundException(AccountConstant.NOT_FOUND));
+  }
+
+  private void sendVerificationMail(String receiver, String token) {
+    String link = baseUrl + verifyUrl + token;
+    byte[] contentBytes = fileService.readHtml(FilePathConstant.VERIFICATION_EMAIL_HTML);
+    String content = new String(contentBytes).replace(HtmlConstant.HREF, link);
+
+    emailService.send(receiver, content);
```

Comment:
Please, attach a screenshot with received email to the PR!

Reply:
Is this connected with the story for edit user profile somehow?

Reply:
> Is this connected with the story for edit user profile somehow?

This part is for sending verification mail.